crates/orderbook-commons/src/price.rs

```diff
@@ -0,0 +1,137 @@
+use crate::Order;
+use crate::ToPrimitive;
+use rust_decimal::Decimal;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::HashMap;
+use trade::ContractSymbol;
+use trade::Direction;
+
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct Price {
+    pub bid: Option<Decimal>,
+    pub ask: Option<Decimal>,
+}
+
+pub type Prices = Box<HashMap<ContractSymbol, Price>>;
+
+/// Best prices across all current orders for given ContractSymbol in the orderbook
+/// Taken orders are not included in the average
+pub fn best_current_price(current_orders: &[Order]) -> Prices {
+    let mut prices = HashMap::new();
+    let mut add_price_for_symbol = |symbol| {
+        prices.insert(
+            symbol,
+            Price {
+                bid: best_bid_price(current_orders, symbol),
+                ask: best_ask_price(current_orders, symbol),
+            },
+        );
+    };
+    add_price_for_symbol(ContractSymbol::BtcUsd);
+    Box::new(prices)
```

Comment:
Why does it have to be box?

Reply:
I thought I would be smart by ensuring it's on the heap, but after reading the docs, `HashMap` is *always* on the heap regardless. 🙈 
removed.